client/src-js/ac.js

```diff
@@ -0,0 +1,159 @@
+const HISTORY_SIZE = 20
+const HISTORY_LIST_SIZE = 4
+const DB_ID = "search_history"
+const WRAPPER_CLASS = "search_class"
+const HOVER_COLOR = '#f4ada4'
+const HOVER_COLOR_CANCEL = '#dddddd'
+const BG_COLOR = '#fefefe'
+const PRIMARY = '#ef8374'
+
+function range(n) {
+  let res = []
+  for (let i=0; i<n; i++) {
+    res.push(i)
+  }
+  return res
+}
+
+function getHistory() {
+  try {
+    return JSON.parse(localStorage.getItem(DB_ID))
```

Comment:
You need something like `return JSON.parse(localStorage.getItem(DB_ID)) || []` to guard this function from returning `null` before any history is saved in storage